app/models/calendar_event.rb

```diff
@@ -14,6 +14,7 @@ class CalendarEvent
   attr_accessor :completion_status
   attr_accessor :time_status
   attr_accessor :num_of_logs
+  attr_accessor :has_precedent
```

Comment:
Boolean accessors like this should always have a `?` at the end. So you could either define it as a regular method or you could add `alias_method :has_precedent?, :has_precedent`. I recommend the latter as it's less code and you'll need the non `?` version in the serializer.